ansible/playbooks/roles/upgrade/tasks/kubernetes.yml

```
@@ -12,14 +12,26 @@
   import_tasks: kubernetes/get-kubelet-version.yml # sets kubelet_version
   delegate_to: "{{ groups.kubernetes_master[0] }}"
 
+- name: Check for upgrade flag file
+  stat:
+    path: "{{ kubernetes.upgrade_state_file_path }}"
+  register: lock_file_status
+
 - name: Upgrade masters then nodes
   vars:
     version: "{{ ver }}"
     cni_version: "{{ cni_ver }}"
   block:
     - name: Upgrade masters
-      when: cluster_version is version('v' + version, '<=')
+      when: >
+        lock_file_status.stat.exists or
+        cluster_version is version('v' + version, '<')
       block:
+        - name: Create upgrade flag file
+          file:
+            path: "{{ kubernetes.upgrade_state_file_path }}"
+            state: touch
```

```suggestion
        - name: Create upgrade state file
          copy:
            dest: "{{ kubernetes.upgrade_state_file_path }}"
            content: Upgrade started
            mode: u=rw,g=r,o=
```
`touch` will update existing file each run so Ansible will report state as `changed`. `copy` will report `ok` when file already exists.

---

nice one, I will remember that!